### ISSUE TITLE
Firefox esr: crash on some web pages

### ISSUE THREAD

**Issue by tristelune1** [kisslinux/repo#136]
Hi,

I have updated `firefox` today and now I'm using the esr version. Unfortunately it crashed on some web pages. Does somebody have the same problem? For example it crashes sometimes on www.rts.ch . Do I have something to do ? 

Thank you!

**Comment by dylanaraps** [comment]
I've also heard reports about this in the IRC channel though I can't personally reproduce the issue.

Are you able to send me a log from Firefox *after* it crashes?

1) `firefox > ~/file 2>&1`
2) Make firefox crash.
3) Upload `~/file`.

**Comment by tristelune1** [comment]
I have no add-on installed. I did only the following change in `about:config`

```
layers.acceleration.disabled      modified    boolean true
layers.acceleration.force-enabled     modified boolean true
```

I set both variables on true. Otherwise I have no hardware acceleration. There was no problem before the change to the esr version. At the moment I have `firefox-bin`, but with the compiled version there is also the same problem. 

I get the following if I load a website which crash:

```
JavaScript error: resource:///modules/sessionstore/SessionStore.jsm, line 1325: uncaught exception: 2147746065
1579065465068	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/fxmonitor@mozilla.org
1579065465068	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/screenshots@mozilla.org
1579065465068	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/webcompat-reporter@mozilla.org
1579065465068	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/webcompat@mozilla.org
JavaScript error: resource:///modules/BrowserGlue.jsm, line 1829: TypeError: addon is null
[Parent 2371, Gecko_IOThread] WARNING: pipe error (44): Connection reset by peer: file /root/.cache/kiss/build-18457/firefox/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 358
[Parent 2371, Gecko_IOThread] WARNING: pipe error: Broken pipe: file /root/.cache/kiss/build-18457/firefox/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 727

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x53001B,name=PHttpChannel::Msg_DeleteSelf) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x1E008F,name=PBrowser::Msg_Destroy) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x1E0074,name=PBrowser::Msg_StopIMEStateManagement) Channel error: cannot send/recv

###!!! [Parent][MessageChannel] Error: (msgtype=0x1E008F,name=PBrowser::Msg_Destroy) Channel error: cannot send/recv
[Parent 2371, Gecko_IOThread] WARNING: pipe error (123): Connection reset by peer: file /root/.cache/kiss/build-18457/firefox/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 358
```

Hmmm, do you have an idea why a file in my `root` directory is used: `/root/.cache/kiss/build-18457/firefox/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 358`  ?

I don't start `firefox` as root. 

Interesting, if I start in safe-mode, I can load the website, but there are some warnings and errors:

```
JavaScript error: resource:///modules/sessionstore/SessionStore.jsm, line 1325: uncaught exception: 2147746065
1579065517037	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/fxmonitor@mozilla.org
1579065517037	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/screenshots@mozilla.org
1579065517038	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/webcompat-reporter@mozilla.org
1579065517038	addons.xpi	WARN	Can't get modified time of /usr/lib/firefox/browser/features/webcompat@mozilla.org
JavaScript error: resource://gre/modules/UrlClassifierListManager.jsm, line 69: NS_ERROR_XPC_GS_RETURNED_FAILURE: ServiceManager::GetService returned failure code:
JavaScript error: resource://gre/modules/SafeBrowsing.jsm, line 272: NS_ERROR_XPC_GS_RETURNED_FAILURE: ServiceManager::GetService returned failure code:
JavaScript error: resource:///modules/BrowserGlue.jsm, line 1829: TypeError: addon is null
JavaScript warning: https://www.raiffeisen.ch/content/dam/scripts/analytics/www/launch-main-www.js, line 2: Error: WebGL warning: <SetDimensions>: Failed to create WebGL context: WebGL is currently disabled.
JavaScript warning: https://www.raiffeisen.ch/content/dam/scripts/analytics/www/launch-main-www.js, line 2: Error: WebGL warning: <SetDimensions>: Failed to create WebGL context: WebGL is currently disabled.
```

**Comment by dylanaraps** [comment]
I tracked down this comment in the Gentoo issue tracker:

```
 Alexander Tsoy 2019-09-18 23:55:19 UTC

For me removing "addonStartup.json.lz4" from profile fixed this problem on several systems.
```

Please try it and let me know if this fixes the issue. I'll keep digging around regardless.

**Comment by dylanaraps** [comment]
If the above fix doesn't work, something else to try is disabling the multi-process windows:

```
    browser.tabs.remote.autostart = false
    browser.tabs.remote.autostart.2 = false 
```

(In `about:config`. You should restart Firefox afterwards)

**Comment by dylanaraps** [comment]
This is the same issue I believe: https://github.com/void-linux/void-packages/issues/12885

**Comment by dylanaraps** [comment]
Another question for those with crashes: Did you transfer your existing profile or start a new one?

**Comment by tristelune1** [comment]
Thank you, for your help! First I started a new profile. I can't test anymore, but in order to start firefox I had to create a new profile. 

If I delete `addonStartup.json.lz4` it doesn't fix the issue. The file is recreated at startup, so it's only deleted for the actual session. But:

```
browser.tabs.remote.autostart = false
```

Fix the problem! :+1:. There is no 
```
browser.tabs.remote.autostart.2 = false
```
 in my configuration. If it's ok, I'll test firefox again tomorrow to be sure it's alright.
